lib/analytics/campaign_csv_builder.rb

```diff
@@ -44,6 +45,11 @@ def revisions_to_csv
     CSV.generate { |csv| csv_data.each { |line| csv << line } }
   end
 
+  def wikidata_to_csv
+    courses = @campaign.courses.joins(:course_stat)
+    CourseWikidataCsvBuilder.new(courses).generate_csv
```

Comment:
I'd prefer this to follow the same pattern as the other methods like `articles_to_csv`, building an aggregate CSV by looping through the courses in the campaign, and adding a row of wikidata stats for each course.

Making `CourseWikidataCsvBuilder` handle the cases of one as well as multiple courses will make this harder to reason about and change, I think.

The problem with that approach is that, unlike the other CSV builders, the output for the Wikidata one isn't a format that is suitable to that aggregation strategy. I think the best way around that will be to change the output format of the Wikidata CSV builder. Currently, it uses headers of `revision_type, count` with one row per revision type. The alternative would be to have a column for each revision type, so that the campaign version could have one row per course.

Reply:
I went with the join so I do not have to loop, but I can change to the `articles_to_csv` pattern.
I also will make 2 methods: one `CourseWikidataCsvBuilder` for courses and another one for campaign.

So, if I understand well, your format proposal for campaign is:
claims created, claims changed, claims removed, etc. course // header
4,2,1, xx,x,"This course name"  // lines of data for one course of a campaign
2,4,5, xx, xxx "This other course name" // same

Also, do you want me to change `CourseWikidataCsvBuilder` so that it will be the same format, but with just one line ?

Reply:
Yes, that's correct, except that `course name` should be the first column.